### ISSUE TITLE
Metadata fields on stream and error outputs

### ISSUE THREAD

**Issue by rgbkrk** [jupyter/nbformat#84]
I'd like to support metadata fields on all the output types. The ones that can't currently are `stream` and `error`.

/cc @craigcitro, who I talked to about this yesterday

**Comment by blink1073** [comment]
👍

**Comment by craigcitro** [comment]
Yes, this would be really helpful for us, too. (More helpful than my slow reply would suggest!)

In particular, in Colaboratory, we use metadata on `stream` outputs, in order to do various forms of output control. We actually didn't realize at first that this was invalid, so we do some gyrations to always ensure this metadata is dropped when writing notebooks out to disk.

**Comment by rgbkrk** [comment]
Thinking on this one after a nice week with the Jupyter devs, how would we merge metadata (if merging streams, like is done in the notebook).

**Comment by rgbkrk** [comment]
@craigcitro -- how does `metadata` get combined if multiple `stdout` messages come in (that can be "merged" into one, since it's a terminal representation).

**Comment by craigcitro** [comment]
At least for colab, we only merge if the messages have identical metadata.
